wpimath/src/main/java/edu/wpi/first/math/geometry/Rectangle2d.java

```diff
@@ -0,0 +1,245 @@
+   * @return The rotated rectangle.
+   */
+  public Rectangle2d rotateBy(Rotation2d other) {
+    return new Rectangle2d(m_center.rotateBy(other), m_xWidth, m_yWidth);
+  }
+
+  /**
+   * Checks if a point is intersected by the rectangle's perimeter.
+   *
+   * @param point The point to check.
+   * @return True, if the rectangle's perimeter intersects the point.
+   */
+  public boolean intersectsPoint(Translation2d point) {
+    // Move the point into the rectangle's coordinate frame
+    point = point.minus(m_center.getTranslation());
+    point = point.rotateBy(m_center.getRotation().unaryMinus());
+
+    if (Math.abs(point.getX()) == m_xWidth / 2.0) {
+      // Point rests on left/right perimeter
+      return Math.abs(point.getY()) <= m_yWidth / 2.0;
+    } else if (Math.abs(point.getY()) == m_yWidth / 2.0) {
+      // Point rests on top/bottom perimeter
+      return Math.abs(point.getX()) <= m_xWidth / 2.0;
+    }
+
+    return false;
+  }
+
+  /**
+   * Checks if a point is contained within the rectangle. This is inclusive, if the point lies on
+   * the perimeter it will return true.
+   *
+   * @param point The point to check.
+   * @return True, if the rectangle contains the point or the perimeter intersects the point.
+   */
+  public boolean containsPoint(Translation2d point) {
+    // Rotate the point into the rectangle's coordinate frame
+    point = point.rotateAround(m_center.getTranslation(), m_center.getRotation().unaryMinus());
+
+    // Check if within bounding box
+    return point.getX() >= (m_center.getX() - m_xWidth / 2.0)
+        && point.getX() <= (m_center.getX() + m_xWidth / 2.0)
+        && point.getY() >= (m_center.getY() - m_yWidth / 2.0)
+        && point.getY() <= (m_center.getY() + m_yWidth / 2.0);
+  }
+
+  /**
+   * Returns the distance between the perimeter of the rectangle and the point.
+   *
+   * @param point The point to check.
+   * @return The distance (0, if the point is contained by the rectangle)
+   */
+  public double distanceToPoint(Translation2d point) {
+    if (containsPoint(point)) {
+      return 0.0;
+    }
+
+    // Move the point into the rectangle's coordinate frame
+    point = point.minus(m_center.getTranslation());
+    point = point.rotateBy(m_center.getRotation().unaryMinus());
+
+    // Find x and y distances
+    double dx = Math.max(0.0, Math.abs(point.getX()) - m_xWidth / 2.0);
+    double dy = Math.max(0.0, Math.abs(point.getY()) - m_yWidth / 2.0);
+
+    // Distance formula
+    return Math.hypot(dx, dy);
+  }
+
+  /**
+   * Returns the nearest point that is contained within the rectangle.
+   *
+   * @param point The point that this will find the nearest point to.
+   * @return A new point that is nearest to {@code point} and contained in the rectangle.
+   */
+  public Translation2d findNearestPoint(Translation2d point) {
+    // Check if already in rectangle
+    if (containsPoint(point)) {
+      return new Translation2d(point.getX(), point.getY());
```

Comment:
Out of curiosity, why not just return `point`? The geometry classes are immutable, so it should be fine to just return the original.